src/ess/diffraction/powder/correction.py

```diff
@@ -44,3 +49,68 @@ def merge_calibration(*, into: sc.DataArray, calibration: sc.Dataset) -> sc.Data
         )
     out.masks['calibration'] = calibration['mask'].data
     return out
+
+
+def lorentz_factor(data: sc.DataArray) -> sc.DataArray:
+    """Compute the ToF powder diffraction Lorentz factor.
+
+    This function uses this definition:
+
+    .. math::
+
+        L = d^4 \\sin\\theta
+
+    where :math:`d` is the d-spacing, :math:`\\theta` is half the scattering angle
+    (note the definitions in
+    https://scipp.github.io/scippneutron/user-guide/coordinate-transformations.html).
+
+    The Lorentz factor as defined here is suitable for correcting time-of-flight data
+    expressed in wavelength or d-spacing.
+    It follows the definition used bty GSAS-II, see page 140 of
+    https://subversion.xray.aps.anl.gov/EXPGUI/gsas/all/GSAS%20Manual.pdf
+
+    Parameters
+    ----------
+    data:
+        Input data with coordinates ``two_theta`` and ``dspacing``.
+
+    Returns
+    -------
+    :
+        Powder Lorentz factor.
+    """
+
+    def f(dspacing: sc.Variable, two_theta: sc.Variable) -> sc.Variable:
+        return dspacing**4 * sc.sin(two_theta / 2)
+
+    aux = data.transform_coords(
+        'lorentz_factor', {'lorentz_factor': f}, rename_dims=False, quiet=True
+    )
```

Comment:
This entire function is odd, it seems `transform_coords` is only used to extract either a binned or regular coord, but then we have to have the special handling below? Shouldn't we either to a single `transform_coords` that also computes `two_theta`, or not have this helper at all?